src/components/Notifications/Notifications.tsx

```diff
@@ -0,0 +1,56 @@
+"use client";
+
+import * as Toast from "@radix-ui/react-toast";
+import type { ReactNode } from "react";
+import { Notification } from "./Notification";
+import styles from "./Notifications.module.css";
+import { useNotifications } from "./NotificationsContext";
+
+export function Notifications() {
+  const { notifications } = useNotifications();
+
+  return (
+    <NotificationsToastProvider>
+      {notifications.map((notification) => {
+        const { hash, chainId } = notification;
+
+        if (notification.type === "assert") {
+          const { claim } = notification;
+          return (
+            <Notification
+              type="assert"
+              key={hash}
+              claim={claim}
+              hash={hash}
+              chainId={chainId}
+            />
+          );
+        }
+        const { formattedAmount, currencySymbol } = notification;
+        return (
+          <Notification
+            type="approve"
```

Comment:
looking at this again, it seems like you might as well just define different notification types and switch them in here, rather than doing the switch once here and then in the notification component, ie create: `NotificationApprove`, `NotificationAssert` and just return them here, removing the more complex `Notification` component